paperqa/agents/tools.py

```diff
@@ -1,196 +1,136 @@
-        """
+        """  # noqa: E501,W505
+        # Convert to date range (e.g. 2022-2022) if date is present
+        year = (
+            f"{min_year if min_year else ''}-{max_year if max_year else ''}"  # noqa: FURB110
+            if (min_year or max_year)
+            else None
+        )
         # get offset if we've done this search before (continuation of search)
         # or mark this search as new (so offset 0)
-        logger.info(f"Starting paper search for '{query}'.")
-        search_key = query
-        if search_key in self.previous_searches:
+        search_key = query, year
+        try:
             offset = self.previous_searches[search_key]
-        else:
+        except KeyError:
             offset = self.previous_searches[search_key] = 0
 
-        # Preprocess inputs to make ScrapeRequest
-        keywords = query.replace('"', "")  # Remove quotes
-        year: str | None = None
-        last_word = keywords.split(" ")[-1]
-        if re.match(r"\d{4}(-\d{4})?", last_word):
-            keywords = keywords[: -len(last_word)].removesuffix(",").strip()
-            if self.search_type[1]:
-                year = last_word
-                if "-" not in year:
-                    year = year + "-" + year  # Convert to date range (e.g. 2022-2022)
-        index = await get_directory_index(
-            settings=self.shared_state.settings,
-        )
-
+        logger.info(f"{self.TOOL_FN_NAME} starting for query {query!r}.")
+        index = await get_directory_index(settings=self.settings)
         results = await index.query(
-            keywords,
-            top_n=self.shared_state.settings.agent.search_count,
+            query,
+            top_n=self.settings.agent.search_count,
             offset=offset,
             field_subset=[f for f in index.fields if f != "year"],
         )
+        logger.info(
+            f"{self.TOOL_FN_NAME} for query {query!r} returned {len(results)} papers."
+        )
 
-        logger.info(f"Search for '{keywords}' returned {len(results)} papers.")
         # combine all the resulting doc objects into one and update the state
         # there's only one doc per result, so we can just take the first one
         all_docs = []
         for r in results:
             this_doc = next(iter(r.docs.values()))
             all_docs.append(this_doc)
-            await self.shared_state.docs.aadd_texts(
+            await state.docs.aadd_texts(
                 texts=r.texts,
                 doc=this_doc,
-                settings=self.shared_state.settings,
-                embedding_model=self.shared_state.embedding_model,
+                settings=self.settings,
+                embedding_model=self.embedding_model,
             )
 
-        status = await self.shared_state.get_status()
-
-        logger.info(status)
-
+        status = state.status
+        logger.info(f"{self.TOOL_FN_NAME} new status {status}.")
```

Comment:
same here let's not do "new status" each line like this, looks much cleaner with: `Status: Paper Count=5 | Relevant Papers=0 | Current Evidence=0 | Current Cost=$0.00.`, and I think new status is implied.

Reply:
Alright, reverted